apps/drec-ui/src/api/user.ts

```diff
@@ -79,3 +98,165 @@ export const useUser = (options?: UseQueryOptions<AxiosResponse<UserDTO>, Error>
         logout
     };
 };
+
+export const useApiResendConfirmationEmail = () => {
+    const { isLoading, error, isError, isSuccess, status, mutateAsync } =
+        useUserControllerReSendEmailConfirmation();
+
+    return {
+        status,
+        isLoading,
+        isSuccess,
+        isError,
+        error,
+        submitHandler: useCallback((): void => {
+            mutateAsync().then(
+                () => {
+                    showNotification('Email was resent', NotificationTypeEnum.Success);
+                },
+                () => {
+                    showNotification('Cannot resed email', NotificationTypeEnum.Error);
+                }
+            );
+        }, [])
+    };
+};
+
+export const useApiUpdateUserAccountPassword = () => {
+    const { logout } = useUser();
+    const queryClient = useQueryClient();
+    const userQueryKey = getUserControllerMeQueryKey();
+
+    const { mutate } = useUserControllerUpdateOwnPassword();
+
+    const submitHandler = (values: TUpdateUserPasswordFormValues) => {
+        const user: UserDTO = queryClient.getQueryData(userQueryKey);
+        const { status } = user;
+        if (status !== UserStatus.Active) {
+            return showNotification(
+                `Only active users can perform this action. Your status is ${user.status}`,
+                NotificationTypeEnum.Error
+            );
+        }
+        mutate(
+            { data: values },
+            {
+                onSuccess: () => {
+                    showNotification(
+                        'User password updated successfully. Please log in using new password',
+                        NotificationTypeEnum.Success
+                    );
+                    logout();
+                },
+                onError: (error: any) => {
+                    showNotification(
+                        `Error while updating user password:
+              ${error?.response?.data?.message || ''}
+              `,
+                        NotificationTypeEnum.Error
+                    );
+                }
+            }
+        );
+    };
+
+    return {
+        submitHandler
+    };
+};
+
+export const useApiUpdateUserAccountEmail = () => {
+    const { mutate } = useUserControllerUpdateOwnProfile();
+    const { logout } = useUser();
+
+    const queryClient = useQueryClient();
+    const userQueryKey = getUserControllerMeQueryKey();
+
+    const submitHandler = (values: Pick<UpdateUserProfileDTO, 'email'>, resetForm: any) => {
```

Comment:
Missing typing for `resetForm`

Reply:
Should also be fixed in origin - updateUserEmail.ts